sklearn/inspection/_partial_dependence.py

```diff
@@ -271,17 +292,43 @@ def partial_dependence(estimator, X, features, *, response_method='auto',
         Please see :ref:`this note <pdp_method_differences>` for
         differences between the 'brute' and 'recursion' method.
 
+    kind : {"legacy", "average", "individual", "both"}, default="legacy"
+        Whether to plot the partial dependence averaged across all the samples
+        in the dataset or one line per sample or both individual lines and the
+        average dependence at the same time.
+
+        - kind="legacy"/"average" results in the traditional PD plot;
+        - kind="individual" results in the ICE plot.
+
+       Note that the fast method="recursion" option is only available for
+       kind="average". Plotting individual dependencies requires using the
+       slower method="brute" option.
+
+        .. versionadded:: 0.23
+
     Returns
     -------
-    averaged_predictions : ndarray, \
+    predictions : ndarray or Bunch
+        - kind='legacy', ndarray
+            shape (n_outputs, len(values[0]), len(values[1]), ...)
+            The predictions for all the points in the grid, averaged over all
+            samples in X (or over the training data if ``method`` is
+            'recursion').
+        - kind='individual', kind='average' or kind='both', Bunch
+            Key 'individual', ndarray
```

Comment:
I have not got much time right now to walk through it with you, but this will render quite poorly as you are not using features of restructured text. You can ordinarily see how it renders by looking at the *ci/circleci: doc artifact* check; but it seems that with the latest commit it failed to build.